### PR TITLE
UI Components: Roadmap entry for missing high priority components

### PR DESCRIPTION
**Pull request by Amstutz**
This is a proposal on how to list missing UI Components in our Roadmap. Important note, the titles are only title for the roadmap entries, part of the work when tackling such an entry is to find a good name for the new Components. Also, the description provided is only meant to give a rough definition of what those components could hold and could be considered or proof challenging when working on the definitions. Feel free to propose alternative concepts that do not exactly match the scope given here.

### CODE REVIEW

File: src/UI/ROADMAP.md [hunk 1]
```diff
@@ -12,12 +12,70 @@ are explained in [Usage](#usage).
...
+The UI-Framework attempts to be the source for all visual elements in ILIAS and
+thus supersede the current templating. The challenge is two-fold: on the one hand
+the required elements need to be implemented in the UI-framework, on the other
+hand the Components need to use the UI-framework for their actual rendering. 
+
+Not all Components have the same priority. Highest on our list are Components that 
+are needed on every screen to render ILIAS with an empty content section (such as Tabs). 
+Next are Components, that are needed in many use cases and therefore have a very high
+grade of re-use (such as the Toolbar). Due to major work done for ILIAS 6, such as the Standard Layout 
+Component, Meta Bar and Main Bar Why are now able to promote further Components to our list of short
```
Comment (kergomard) [inline src/UI/ROADMAP.md]:
```suggestion
Component, Meta Bar and Main Bar we are now able to promote further Components to our list of short 
```

File: src/UI/ROADMAP.md [hunk 2]
```diff
@@ -12,12 +12,70 @@ are explained in [Usage](#usage).
...
+content (currently provided as array of Legacy Components). Most Probably it should be able to hold the title section 
+(not yet part of the UI Components, see below), the Tabs (not yet Part of the UI Components, see below) and the
+Inner Content holding the workspace for the current context (not yet Part of the UI Components, see below).
+
+Note; One important aspect here, will be to clarify at some point the relation to the [Global Screen](../GlobalScreen). 
+
+#### Title Section (advanced, variable)
+This Component will probably hold the Icon, title, description and the actions (maybe along with the used glyphs) of the 
+current context. Note that a major part of the work for this components will be to setup a comprehensive set of rules on 
+when to provide and Icon, restrictions of the Title (lengths, nouns vs verbs etc.), restrictions of the description
```
Comment (kergomard) [inline src/UI/ROADMAP.md]:
```suggestion
when to provide an Icon, restrictions of the Title (lengths, nouns vs verbs etc.), restrictions of the description 
```

File: src/UI/ROADMAP.md [hunk 3]
```diff
@@ -12,12 +12,70 @@ are explained in [Usage](#usage).
...
+
+Note; One important aspect here, will be to clarify at some point the relation to the [Global Screen](../GlobalScreen). 
+
+#### Title Section (advanced, variable)
+This Component will probably hold the Icon, title, description and the actions (maybe along with the used glyphs) of the 
+current context. Note that a major part of the work for this components will be to setup a comprehensive set of rules on 
+when to provide and Icon, restrictions of the Title (lengths, nouns vs verbs etc.), restrictions of the description 
+(lengths, when to use etc.) and nature, amount of the actions etc. Note that there is pre-existing work on those 
+subjects: [Feature Wiki](https://docu.ilias.de/goto_docu_wiki_wpage_6080_1357.html). 
+However, this is not decided upon and most certainly up for discussion.
```
Comment (kergomard) [inline src/UI/ROADMAP.md]:
```suggestion
However, this has not been decided yet and is thus most certainly up for discussion.
```

File: src/UI/ROADMAP.md [hunk 4]
```diff
@@ -12,12 +12,70 @@ are explained in [Usage](#usage).
...
+
+Note; One important aspect here, will be to clarify at some point the relation to the [Global Screen](../GlobalScreen). 
+
+#### Tabs and Sub Tabs (advanced, variable)
+Note that a major part of the work for this Components will be to setup a comprehensive set of rules on the naming of 
+Tabs and Sub Tabs (noun vs verbs, length, amount of words etc.) and rules for the usage of Tabs vs Sub Tabs vs Sections
+in Forms shown in Tabs. Also, one would have to look into the issue that currently "<-- Back" actions are mixed into 
+the Tabs. We will need to decide, whether we will still use this concept in future.
```
Comment (kergomard) [inline src/UI/ROADMAP.md]:
```suggestion
the Tabs. We will need to decide, whether we will still use this concept in the future.
```

**Review by kergomard — COMMENTED** [review]
Thank you very much for this! Reading through it I just found a few typos, I tried to correct inline.

**Comment by Amstutz** [comment]
Thx @swiniker for reading through and the spotting of typos. Fixed.

**Comment by Amstutz** [comment]
Note, @chfsx will have a first look at the listed components, thx a lot @chfsx !

**Review by klees — APPROVED** [review]
Thanks @Amstutz, sounds plausible. Let's add this =)

**Pull request merged**